onnxruntime/core/providers/cuda/tensor/split_impl.cu

```diff
@@ -157,5 +157,122 @@ Status SplitImpl(cudaStream_t stream, const size_t element_size, const int block
   return Status::OK();
 }
 
+template <typename T>
+__device__ __forceinline__ void _vec_copy_data(
+    void* output_ptr,
+    const void* input_ptr,
+    const int64_t size_in_byte) {
+  auto output_ptr_vec = reinterpret_cast<T*>(output_ptr);
+  auto input_ptr_vec = reinterpret_cast<const T*>(input_ptr);
+  auto new_size = size_in_byte / sizeof(T);
+  for (int i = threadIdx.x; i < new_size; i += blockDim.x) {
+    output_ptr_vec[i] = input_ptr_vec[i];
+  }
+  auto remain = size_in_byte % sizeof(T);
+  auto output_ptr_r = reinterpret_cast<char*>(output_ptr_vec + new_size);
+  auto input_ptr_r = reinterpret_cast<const char*>(input_ptr_vec + new_size);
+  for (int i = threadIdx.x; i < remain; i += blockDim.x) {
+    output_ptr_r[i] = input_ptr_r[i];
+  }
+}
+
+template <typename T>
+__device__ __forceinline__ void _block_copy_data(T* out, const T* input, int size) {
+  auto copy_size_in_byte = size * sizeof(T);
+
+  auto select = [](int value) {
+    if (value % 16 == 0) {
+      return 16;
+    } else if (value % 8 == 0) {
+      return 8;
+    } else if (value % 4 == 0) {
+      return 4;
+    } else if (value % 2 == 0) {
+      return 2;
+    } else {
+      return 1;
+    }
+  };
+  // cast the input and output ptr into long value, and check the alignment size of there pointers.
+  auto input_v = reinterpret_cast<size_t>(input);
+  auto out_v = reinterpret_cast<size_t>(out);
+  // select the min alignment of input and output pointer
+  auto vec_size = std::min(select(input_v), select(out_v));
```

Comment:
Not only the pointers, need also select(copy_size_in_byte) here. For example, if we copy 4 bytes, we cannot return 16.